### PR TITLE
Fixing issue #121

### PR DESCRIPTION
**Pull request by alphaville**
- introduced `InnerProblemStatus` in `AlmOptimizer`
- added unit test

### CODE REVIEW

File: src/alm/alm_optimizer.rs [hunk 1]
```diff
@@ -15,6 +15,26 @@ const DEFAULT_INFEAS_SUFFICIENT_DECREASE_FACTOR: f64 = 0.1;
 const DEFAULT_INITIAL_TOLERANCE: f64 = 0.1;
 const SMALL_EPSILON: f64 = std::f64::EPSILON;
 
+/// Internal/private structure used by method AlmOptimizer.step
+/// to return some minimal information about the inner problem
+struct InnerProblemStatus {
```
Comment (korken89) [inline src/alm/alm_optimizer.rs]: We should probably add `#[derive(Debug, Copy, Clone)]` for when purple want to help contribute and start printing :)

Comment (alphaville) [reply src/alm/alm_optimizer.rs]: This structure is not public. It's only used within `AlmOptimizer`.

Comment (korken89) [reply src/alm/alm_optimizer.rs]: What happens in you do `println!("{#?}", alm_optimizer)`?
I think you will get errors on debug not being implemented, but maybe not.

Not sure if we test this?

Comment (alphaville) [reply src/alm/alm_optimizer.rs]: `AlmOptimizer` doesn't hold instances of or references to `InnerProblemStatus` - this is just returned by method `step`. However, `AlmOptimizer` has an instance of `AlmProblem` which is in turn has fields that need to satisfy the trait constraint:

```rust
MappingAlm: Fn(&[f64], &mut [f64]) -> Result<(), SolverError>
```

therefore, as far as I know, `Debug` cannot be derived. However, if someone would like to print some meaningful information, they should print `AlmCache`, which contains all information used at every iteration of the algorithm (both for the outer and the inner loops).

File: open-codegen/opengen/templates/optimizer_cargo.toml.template [hunk 2]
```diff
@@ -18,7 +18,9 @@ publish=false
 
 
 [dependencies]
-optimization_engine = "{{open_version or '0.6.0'}}"
+# optimization_engine = "{{open_version or '0.6.0'}}"
```
[the remark this answers is not in the document]
Comment (alphaville) [reply open-codegen/opengen/templates/optimizer_cargo.toml.template]: done

File: open-codegen/opengen/templates/tcp_server_cargo.toml.template [hunk 3]
```diff
@@ -21,7 +21,8 @@ publish=false
 
 
 [dependencies]
-optimization_engine = "{{open_version or '0.6.0'}}"
+# optimization_engine = "{{open_version or '0.6.0'}}"
```
[the remark this answers is not in the document]
Comment (alphaville) [reply open-codegen/opengen/templates/tcp_server_cargo.toml.template]: done

**Pull request merged**